commit ea62d025785b7ae9b1e76ee21d345901a7013302 (refs/changes/74/5074/2)
Author: Josh Wilsdon <josh@wilsdon.ca>
Date:   2018-11-20T10:07:08-08:00 (11 months ago)
    
    TRITON-971 CNAPI should not crash when CN has no boot params
    Reviewed by: Orlando Vazquez <orlando@joyent.com>
    Approved by: Orlando Vazquez <orlando@joyent.com>

diff --git a/lib/models/server.js b/lib/models/server.js
index ecb351e..a1da652 100644
--- a/lib/models/server.js
+++ b/lib/models/server.js
@@ -917,7 +917,9 @@ ModelServer.initialValues = function (opts, callback) {
                 function (error, params) {
                     boot_params = params;
 
-                    if (boot_params.kernel_args.rabbitmq_dns) {
+                    if (boot_params && boot_params.kernel_args &&
+                        boot_params.kernel_args.rabbitmq_dns) {
+
                         boot_params.kernel_args.rabbitmq
                             = boot_params.kernel_args.rabbitmq_dns;
                         delete boot_params.kernel_args.rabbitmq_dns;
@@ -946,12 +948,20 @@ ModelServer.initialValues = function (opts, callback) {
         server.reserved = false;
         server.vms = {};
 
-        server.boot_platform = boot_params.platform;
-        server.boot_params = boot_params.kernel_args || {};
-        server.kernel_flags = boot_params.kernel_flags || {};
+        if (boot_params) {
+            server.boot_platform = boot_params.platform;
+            server.boot_params = boot_params.kernel_args;
+            server.kernel_flags = boot_params.kernel_flags;
+            server.default_console = boot_params.default_console;
+            server.serial = boot_params.serial;
+        }
+
+        // set defaults if not set above
+        server.boot_params = server.boot_params || {};
+        server.kernel_flags = server.kernel_flags || {};
+        server.default_console = server.default_console || 'serial';
+        server.serial = server.serial || 'ttyb';
 
-        server.default_console = boot_params.default_console || 'serial';
-        server.serial = boot_params.serial || 'ttyb';
         server.created = (new Date()).toISOString();
 
         callback(null, server);
diff --git a/package.json b/package.json
index 6ed3965..9cfbcb8 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "cnapi",
   "description": "SmartDataCenter Compute Node API",
-  "version": "1.14.0",
+  "version": "1.14.1",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
